cli/lib/kontena/cli/cloud/login_command.rb

```diff
@@ -4,55 +4,56 @@ module Kontena::Cli::Cloud
   class LoginCommand < Kontena::Command
     include Kontena::Cli::Common
 
-    option ['-t', '--token'], '[TOKEN]', 'Use a pre-generated access token'
+    option ['-t', '--token'], '[TOKEN]', 'Use a pre-generated access token', environment_variable: 'KONTENA_ACCOUNT_TOKEN'
     option ['-c', '--code'], '[CODE]', 'Use an authorization code'
     option ['-v', '--verbose'], :flag, 'Increase output verbosity'
+    option ['-f', '--force'], :flag, 'Force reauthentication'
 
     def execute
-      require_relative '../localhost_web_server'
-      require 'launchy'
+      if self.code && self.force?
+        exit_with_error "Can't use --code and --force together"
+      end
 
-      if self.code
-        kontena_account.token ||= Kontena::Cli::Config::Token.new(access_token: self.token, parent_type: :account, parent_name: kontena_account.name)
-        client = Kontena::Client.new(kontena_account.token_endpoint, kontena_account.token)
-        response = nil
-        begin
-          vspinner "Exchanging authorization code to access token" do
-            response = client.exchange_code(self.code)
-            raise Kontena::Errors::StandardError.new(400, 'Code exchange failed') unless response
-          end
-        end
-        if response && response.kind_of?(Hash) && response['access_token']
-          kontena_account.token.access_token = response['access_token']
-          kontena_account.token.refresh_token = response['refresh_token']
-          kontena_account.token.expires_at = response['expires_in'].to_i > 0 ? Time.now.utc.to_i + response['expires_in'].to_i : nil
-          logger.debug "Code exchanged succesfully"
-        else
-          puts "Code exchange failed".colorize(:red)
-          exit 1
-        end
-      elsif self.token.nil?
-        token = kontena_account.token ||= Kontena::Cli::Config::Token.new(parent_type: :account, parent_name: ENV['KONTENA_ACCOUNT'] || 'kontena')
-      elsif self.token
-        kontena_account.token = Kontena::Cli::Config::Token.new(access_token: self.token, parent_type: :account, parent_name: ENV['KONTENA_ACCOUNT'] || 'kontena')
+      if self.token
+        exit_with_error "Can't use --token and --force together" if self.force?
+        exit_with_error "Can't use --token and --code together"  if self.code
+      end
+
+      if !kontena_account.token || !kontena_account.token.access_token || self.token || self.force?
```

Comment:
This could be refactored into method `create_access_token?`, would make code cleaner.
What happens if `self.token` is nil?